blame.c

```diff
@@ -1423,10 +1423,17 @@ static struct blame_origin *find_rename(struct repository *r,
 	struct blame_origin *porigin = NULL;
 	struct diff_options diff_opts;
 	int i;
+	extern int rename_detection_mode;
 
 	repo_diff_setup(r, &diff_opts);
 	diff_opts.flags.recursive = 1;
-	diff_opts.detect_rename = DIFF_DETECT_RENAME;
+	/* 
+	 * Use rename_detection_mode if specified, otherwise default to DIFF_DETECT_RENAME
+	 * For mode values > 0 and < 100, use it as similarity threshold
+	 */
+	diff_opts.detect_rename = (rename_detection_mode == 0) ? 0 : 
+	                           (rename_detection_mode > 0) ? 
+	                           rename_detection_mode : DIFF_DETECT_RENAME;
```

Comment:
This is the only place where you actually use `rename_detection_mode`, which you described elsewhere as a percentage. However, it is used exclusively as a Boolean. You have to study `diff*.c` harder to see how the `--find-renames` value is used there and imitate it here.